src/fsharp/utils/FileSystem.fs

```diff
@@ -164,6 +164,30 @@ type SafeUnmanagedMemoryStream =
         base.Dispose disposing
         x.holder <- null // Null out so it can be collected.
 
+type internal MemoryMappedStream(mmf: MemoryMappedFile, length: int64) = 
+    inherit Stream()
+
+    member private _.Mmf = mmf
+    
+    member _.ViewStream = mmf.CreateViewStream(0L, length, MemoryMappedFileAccess.Read)
```

Comment:
This is a property which is creating a new ViewStream each time it is used

I think you want 

```
let viewStream = mmf.CreateViewStream(0L, length, MemoryMappedFileAccess.Read)
member _.ViewStream = viewStream
```

Reply:
I think we also need to code disposal logic more carefully, specifically always using a `let mutable disposed = false` flag to guard against double disposal.  We need to be more forgiving on double disposal too - no ObjectDisposedExceptions - and try to capture good debug stacks if double disposal does happen

Reply:
> This is a property which is creating a new ViewStream each time it is used
> 
> I think you want
> 
> ```
> let viewStream = mmf.CreateViewStream(0L, length, MemoryMappedFileAccess.Read)
> member _.ViewStream = viewStream
> ```

Yeah, I've been mostly poking around with the stream wrapper, just quickly pushed it to run tests in CI, since my machine is quite slow.